swabbie-aws/src/main/kotlin/com/netflix/spinnaker/swabbie/aws/launchconfigurations/AmazonLaunchConfigurationHandler.kt

```diff
@@ -0,0 +1,198 @@
+import com.netflix.spinnaker.swabbie.events.Action
+import com.netflix.spinnaker.swabbie.exclusions.ResourceExclusionPolicy
+import com.netflix.spinnaker.swabbie.model.AWS
+import com.netflix.spinnaker.swabbie.model.LAUNCH_CONFIGURATION
+import com.netflix.spinnaker.swabbie.model.MarkedResource
+import com.netflix.spinnaker.swabbie.model.WorkConfiguration
+import com.netflix.spinnaker.swabbie.notifications.NotificationQueue
+import com.netflix.spinnaker.swabbie.notifications.Notifier
+import com.netflix.spinnaker.swabbie.orca.OrcaJob
+import com.netflix.spinnaker.swabbie.orca.OrcaService
+import com.netflix.spinnaker.swabbie.orca.OrchestrationRequest
+import com.netflix.spinnaker.swabbie.repository.TaskCompleteEventInfo
+import com.netflix.spinnaker.swabbie.repository.TaskTrackingRepository
+import com.netflix.spinnaker.swabbie.repository.ResourceTrackingRepository
+import com.netflix.spinnaker.swabbie.repository.ResourceStateRepository
+import com.netflix.spinnaker.swabbie.repository.ResourceUseTrackingRepository
+import com.netflix.spinnaker.swabbie.rules.RulesEngine
+import com.netflix.spinnaker.swabbie.utils.ApplicationUtils
+import org.springframework.context.ApplicationEventPublisher
+import org.springframework.stereotype.Component
+import java.time.Clock
+import kotlin.system.measureTimeMillis
+
+@Component
+class AmazonLaunchConfigurationHandler(
+  registry: Registry,
+  clock: Clock,
+  notifier: Notifier,
+  resourceTrackingRepository: ResourceTrackingRepository,
+  resourceStateRepository: ResourceStateRepository,
+  resourceOwnerResolver: ResourceOwnerResolver<AmazonLaunchConfiguration>,
+  exclusionPolicies: List<ResourceExclusionPolicy>,
+  applicationEventPublisher: ApplicationEventPublisher,
+  swabbieProperties: SwabbieProperties,
+  dynamicConfigService: DynamicConfigService,
+  private val rulesEngine: RulesEngine,
+  private val aws: AWS,
+  private val orcaService: OrcaService,
+  private val applicationUtils: ApplicationUtils,
+  private val taskTrackingRepository: TaskTrackingRepository,
+  resourceUseTrackingRepository: ResourceUseTrackingRepository,
+  notificationQueue: NotificationQueue
+) : AbstractResourceTypeHandler<AmazonLaunchConfiguration>(
+  registry,
+  clock,
+  rulesEngine,
+  resourceTrackingRepository,
+  resourceStateRepository,
+  exclusionPolicies,
+  resourceOwnerResolver,
+  notifier,
+  applicationEventPublisher,
+  resourceUseTrackingRepository,
+  swabbieProperties,
+  dynamicConfigService,
+  notificationQueue
+) {
+  override fun deleteResources(markedResources: List<MarkedResource>, workConfiguration: WorkConfiguration) {
+    val application = applicationUtils.determineApp(markedResources.first().resource)
+    val launchConfigurationNames = markedResources
+      .map {
+        it.resourceId
+      }.toSet()
+
+    orcaService.orchestrate(
+      OrchestrationRequest(
+        application = application,
+        job = listOf(
+          OrcaJob(
```

Comment:
do we need to introduce a wait like how we do for images ? not sure the # of launch configs we will start deleting

Reply:
Let's collect some metrics first.